test/gurka/test_search_filter.cc

```diff
@@ -182,3 +185,841 @@ TEST_F(SearchFilter, ExcludeRamp) {
+                                      [](baldr::GraphReader& reader, baldr::TrafficTile& tile,
+                                         int index, baldr::TrafficSpeed* current) -> void {
+                                        SetLiveSpeed(current, default_speed);
+                                      });
+  }
+
+  virtual void SetUp() {
+    set_default_speed_on_all_edges();
+  }
+
+  virtual void TearDown() {
+    set_default_speed_on_all_edges();
+  }
+};
+
+gurka::map ExcludeClosuresOnWaypoints::closure_map = {};
+const int ExcludeClosuresOnWaypoints::default_speed = 36;
+const std::string ExcludeClosuresOnWaypoints::tile_dir =
+    "test/data/traffic_exclude_closures_on_waypoints";
+std::shared_ptr<baldr::GraphReader> ExcludeClosuresOnWaypoints::reader;
+
+/*
+ *  Tests search_filter.exclude_closures at departure
+ */
+TEST_P(ExcludeClosuresOnWaypoints, ExcludeClosuresAtDeparture) {
+  std::string costing = std::get<0>(GetParam());
+  std::string date_type = std::get<1>(GetParam());
+  std::string costing_speed_type =
+      (boost::format("/costing_options/%s/speed_types/0") % costing).str();
+
+  // None of the edges are closed
+  {
+    auto result = gurka::route(closure_map, {"1", "2"}, costing,
+                               {{"/date_time/type", date_type},
+                                {"/date_time/value", "current"},
+                                {costing_speed_type, "current"}},
+                               reader);
+    gurka::assert::osrm::expect_steps(result, {"AB"});
+    gurka::assert::raw::expect_path(result, {"AB", "BC", "CD", "DE"});
+  }
+
+  // AB edge is closed in both directions. Route should avoid AB with
+  // exclude_closures set to true (default) & and use it otherwise
+  {
+    LiveTrafficCustomize close_edge = [](baldr::GraphReader& reader, baldr::TrafficTile& tile,
+                                         int index, baldr::TrafficSpeed* current) -> void {
+      close_single_edge(reader, tile, index, current, "AB", closure_map);
+    };
+    test::customize_live_traffic_data(closure_map.config, close_edge);
+
+    auto result = gurka::route(closure_map, {"1", "2"}, costing,
+                               {{"/date_time/type", date_type},
+                                {"/date_time/value", "current"},
+                                {costing_speed_type, "current"}},
+                               reader);
+    gurka::assert::osrm::expect_steps(result, {"HIC", "CD"});
+    gurka::assert::raw::expect_path(result, {"HIC", "CD", "DE"});
+
+    // Specify search filter to disable exclude_closures at departure
+    const std::string& req_disable_exclude_closures =
+        (boost::format(
+             R"({"locations":[{"lat":%s,"lon":%s,"search_filter":{"exclude_closures":false}},{"lat":%s,"lon":%s}],"costing":"%s", "costing_options": {"%s": {"speed_types":["freeflow","constrained","predicted","current"]}}, "date_time":{"type":"%s", "value": "current"}})") %
```

Comment:
I think you can use gurka::route without pre-generating the request string like you do above

```
auto result = gurka::route(closure_map, {"1", "2"}, costing,
                               {{"/locations/0/search_filter/exclude_closures", false}},
                               {{"/date_time/type", date_type},
                                {"/date_time/value", "current"},
                                {costing_speed_type, "current"}},
                               reader);
```

Reply:
Unfortunately this won't work because, in the `route()` overload accepts `unordered_map <string, string>`, which means that I have to pass-in `false` as `"false"`...and this causes this option to be a string in the resulting json (when what we want is a json boolean type).

This could possibly be fixed by passing in the `rapidjson::Value` variant (which can accommodate different types). But I'll have to test it out and see. For now, I'll keep it as-is.